### PR TITLE
tests: add two tests to SIR

### PR DESCRIPTION
**Pull request by felipehuerta17**
**Branched from the [modules branch](https://github.com/open-sir/open-sir/tree/modules), see #22** 

After a bit of struggle, I started creating unit tests for the SIR model on the file tests/test_sir.py.

To create tests, create a file or write functions in the existing files which start with test_*

To run tests, 
`simplly pipenv run test`

1) **test_sixdays():**
Test predictions against Jupyter notebook case. The results are hard coded and the error is extected to be below the tolerance 1e-4.

2) **Exponential convergence**
A very famous test to the SIR model is to check whether the
one-parameter approximation

I ~ I_0 * exp (alpha-beta) as (S,t) -> (1,0)

Holds. This test implements it with a large population and short time
to ensure numerical consistency.

Note: this only holds for extremely large populations and 0.01s with decent accuracy, so never in life should be used and people fitting exponential models to predict S are extremely wrong.

I am happy to receive comments and suggestions to this test so we can create a good template.

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
This PR depends on #22, and we should not merge until that one.

**Comment by leandrolanzieri** [comment]
@sasalatart is there a way to in circleci to block based on a label?

File: SIR-X.ipynb [hunk 1]
```diff
@@ -541,8 +578,20 @@
    },
    "outputs": [],
    "source": [
-    "# Get the confidence interval through bootstrap\n",
-    "par_ci, par_list = ci_bootstrap(SIR_UK, t_d, I_UK, P_UK, n_iter=1000)"
+    "# Get the confidence interval through random bootstrap\n",
```
Comment (sasalatart) [inline SIR-X.ipynb:581]: Why are there so many changes in non-tests files?

**Comment by sasalatart** [comment]
> #22

Was this branch created from that branch? (`models` branch)

If that is the case, please remember to point the PR to that branch, and only point this branch once the origin branch has been merged, or something like that. If not, then this PR will show changes from the base branch

File: SIR-X.ipynb (hunk 1 above)
Comment (sasalatart) [reply SIR-X.ipynb:581]: Ah, I think it has to do with this: https://github.com/open-sir/open-sir/pull/23#issuecomment-609497860

**Comment by sasalatart** [comment]
> @sasalatart is there a way to in circleci to block based on a label?

What kind of labels? You mean the GitHub label `enhancement` which was added to this PR?

**Comment by felipehuerta17** [comment]
@sasalatart @leandrolanzieri thanks for your comments, this comes from branch `modules` which is trying to merge to master on #22 and I forgot to write that on the description.

**Comment by jia200x** [comment]
Now since #22 got merged, you can continue with this one.

However, after the big squash in #22, if you try to rebase this branch into master it will fail (a lot of commits don't exist anymore).
But `git rebase` is always a friend.

The trick here is to only take the commits that belong to this PR (I guess the first one after 3e16cdacdd2bcb8ec2040134ea1c7b703f0f2353, you probably know better) and put them on top of the current master (that already includes #22).

This pattern can be easily done with the `--onto` feature of git rebase. 
```
git rebase --onto newBase oldBase feature/branch
```

For instance:
```
git checkout master
git pull origin master
git checkout mod/tests
git rebase --onto master FIRST_COMMIT_OF_THIS_PR mod/tests
git push origin mod/tests --force
```

This procedure is well explained [here](https://stackoverflow.com/questions/10853935/change-branch-base)

**Comment by jia200x** [comment]
> What kind of labels? You mean the GitHub label enhancement which was added to this PR?

Yes, we use those in RIOT to block a PR if the "Waiting for PR" label is set. This way, there's no risk of merging a PR without merging the dependencies first

**Comment by sasalatart** [comment]
> > What kind of labels? You mean the GitHub label enhancement which was added to this PR?
> 
> Yes, we use those in RIOT to block a PR if the "Waiting for PR" label is set. This way, there's no risk of merging a PR without merging the dependencies first

Interesting. I think we can access these things with a query to GitHub within the CI (or something like that). I will investigate that during the week and let you know @leandrolanzieri @jia200x

**Comment by felipehuerta17** [comment]
Just to make a note, the first commit is 
6557253

**Comment by jia200x** [comment]
Some comments (for further PRs)
- The first commit doesn't follow the Angular JS convention. 
- Some commits are not atomic (commit 2 also touches the six days function)

Regarding this PR, I think all commits can be squashed into a single "test(model): add initial tests". 
Otherwise I'm fine with the PR

**Comment by felipehuerta17** [comment]
@jia200x suggestions followed and the squash was done manually.

**Comment by jia200x** [comment]
Perfect! Thanks!

**Comment by jia200x** [comment]
```
tests/test_dummy.py .                                                                                                                                                                        [ 33%]
tests/test_sir.py ..                                                                                                                                                                         [100%]

======================================================================================== 3 passed in 0.35s =========================================================================================
```

All good!

**Pull request merged**